docs/internals-mappings.md

```diff
@@ -0,0 +1,428 @@
+
+Mappings between the two different Object Definitions are defined within the Terraform Generator, both [for simple types](https://github.com/hashicorp/pandora/blob/d893d8c6533bbfc3feab00dea5fcf58f880897df/tools/generator-terraform/generator/mappings/assignment_direct.go#L10-L22) and [complex types requiring some form of transformation](https://github.com/hashicorp/pandora/blob/d893d8c6533bbfc3feab00dea5fcf58f880897df/tools/generator-terraform/generator/mappings/assignment_direct_transform.go#L5-L14).
+
+Conversions between Object Definition Types (for example the Schema containing an Integer but the SDK containing a String value) are not supported at this time - but if needed would likely be supported as a new type of Mapping.
+
+Direct Assignment Mappings support the following properties:
+
+* `schemaModelName` - specifies the name of the Terraform Schema Model where this value should be mapped from.
+* `schemaFieldPath` - specifies the path to the Field within the Terraform Schema Model defined in `schemaModelName` which this should be mapped from.
+* `sdkModelName` - specifies the name of the SDK Model where this value should be mapped onto.
+* `sdkFieldPath` - specifies the Path to the Field within the SdkModel where the Schema Field should be mapped onto.
+
+> **Note:** Whilst `schemaFieldPath` and `sdkFieldPath` are named `Path` to allow supporting indexed keys in the future (for example using `AppSettings.SomeSetting` to retrieve the value for the key `SomeSetting` within the Dictionary defined in the `AppSettings` property) - at this point in time these only support top-level fields (e.g. `Name`).
+
+An example of a Direct Assignment Mapping can be seen below:
+
+> **Note:** A more complete example can be seen under `Examples` below.
+
+```json
+{
+  "fieldMappings": [
+    {
+      "type": "DirectAssignment",
+      "directAssignment": {
+        "schemaModelName": "SolarSystemResourceSchema",
+        "schemaFieldPath": "Name",
+        "sdkModelName": "SolarSystem",
+        "sdkFieldPath": "Name"
+      }
+    }
+  ]
+}
+```
+
+### b. `ModelToModel` Field Mappings
+
+ModelToModel Field Mappings - not to be confused with ModelToModel Mappings (below) - allow mapping a Terraform Schema Model onto a field within an SDK Model.
+
+The best example of this is allowing the top-level Terraform Schema Model for a given Terraform Resource to map onto the `properties` object that exists in most payloads in Azure Resource Manager - allowing for fields which exist at the root of the Terraform Schema to be mapped to a nested object.
+
+Each ModelToModel Field Mapping must have an associated ModelToModel mapping, these are used together:
+
+* A ModelToModel Field Mapping is used to output the call to the mapping function itself.
+* A ModelToModel Mapping is used to output the mapping function, the fields which should be mapped are then inferred from the ModelToModel Field Mappings.
```

Comment:
This sounds like they're both doing the same thing: 
`output the call to the mapping function` vs. `output the mapping function`
This might want to be rephrased or more information to distinguish the difference in functionality.